src/tmpfiles/tmpfiles.c

```diff
@@ -2912,16 +3087,37 @@ static int parse_line(
 
         if (!empty_or_dash(age)) {
                 const char *a = age;
+                _cleanup_free_ char *seconds = NULL, *age_by = NULL;
 
                 if (*a == '~') {
                         i.keep_first_level = true;
                         a++;
                 }
 
-                r = parse_sec(a, &i.age);
+                /* Format: "age-by:age"; where age-by is "[abcmABM]+". */
+                if (strchr(a, ':')) {
+                        r = split_pair(a, ":", &age_by, &seconds);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0) {
+                                *invalid_config = true;
+                                return log_syntax(NULL, LOG_ERR, fname, line, r, "Failed to parse age '%s'.", age);
+                        }
+
+                        r = parse_age_by_from_arg(age_by, &i);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0) {
+                                *invalid_config = true;
+                                return log_syntax(NULL, LOG_ERR, fname, line, r, "Invalid age-by '%s'.", age_by);
+                        }
+
+                }
+
+                r = parse_sec((seconds ? seconds : a), &i.age);
```

Comment:
`(seconds ? seconds : a)` → `seconds ?: a`

Reply:
Updated to this https://github.com/systemd/systemd/pull/19440#discussion_r640382584, instead.